drgn/helpers/linux/stackdepot.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) Google LLC
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+"""
+Stack Depot
+-----------
+
+The ``drgn.helpers.linux.stackdepot`` module provides helpers for working with
+the stack depot implementation used by KASAN and other kernel debugging tools.
+"""
+
+from typing import Optional
+
+from drgn import Object, StackTrace, cast, reinterpret
+
+__all__ = ("stack_depot_fetch",)
+
+
+def stack_depot_fetch(handle: Object) -> Optional[StackTrace]:
+    """
+    Returns a stack trace for the given stack handle.
+
+    :param handle: ``depot_stack_handle_t``
+    :return: The stack trace, or ``None`` if not available.
+    """
+    prog = handle.prog_
+    handle_parts = reinterpret("union handle_parts", handle)
+
+    # Renamed in commit 961c949b012f ("lib/stackdepot: rename slab to pool").
+    try:
+        pool = prog["stack_pools"][handle_parts.pool_index]
+    except KeyError:
+        pool = prog["stack_slabs"][handle_parts.slabindex]
+
+    if not pool:
+        return None
+
+    # This has remained the same since the stack depot was introduced in
+    # commit cd11016e5f52 ("mm, kasan: stackdepot implementation. Enable
+    # stackdepot for SLAB"), when it was known as STACK_ALLOC_ALIGN.
+    DEPOT_STACK_ALIGN = 4
+
+    record = cast(
+        "struct stack_record *", pool + (handle_parts.offset << DEPOT_STACK_ALIGN)
+    )
+    trace = list(map(lambda x: record.entries[x], range(record.size)))
+    return prog.stack_trace_from_pcs(trace)
```

Comment:
A list comprehension would be more idiomatic here:
```suggestion
    return prog.stack_trace_from_pcs([record.entries[x] for x in range(record.size)])
```